core/src/main/java/io/questdb/std/str/Utf8SplitString.java

```diff
@@ -53,6 +54,32 @@ public byte byteAt(int index) {
         return Unsafe.getUnsafe().getByte((index < VARCHAR_INLINED_PREFIX_BYTES ? lo1 : lo2) + index);
     }
 
+    @Override
+    public long zeroPaddedSixPrefix() {
+        return Unsafe.getUnsafe().getLong(lo1) & VARCHAR_INLINED_PREFIX_MASK;
+    }
+
+    @Override
+    public long zeroPaddedLongAt(int index) {
+        return index <= size() - Long.BYTES
+                ? Unsafe.getUnsafe().getLong(lo2 + index)
+                : Utf8Sequence.super.zeroPaddedLongAt(index);
+    }
+
+    @Override
+    public boolean equalsAssumingSameSize(Utf8Sequence other) {
+        if (zeroPaddedSixPrefix() != other.zeroPaddedSixPrefix()) {
```

Comment:
Since the size is already assumed to be the same, it is impossible to have a split varchar on LHS and an inlined varchar on RHS.

Reply:
It's impossible currently, but not checked/guaranteed by the code. It may break at any point in future (or even now since there are many `Utf8Sequence` implementations).

Reply:
My rationale is to have it break to indicate a programming mistake. We shouldn't overuse this method for unrrelated tasks.

Reply:
We have a long history of fixing similar `UnsupportedOperationException` ninja bugs, so that's why I'd better be on the safer side even though not all classes implement this method in the most efficient way.